### ISSUE TITLE
Recording at writing if a column is sorted or not, for faster filtering?

### ISSUE THREAD

**Issue by yohplala** [dask/fastparquet#669]
Hi,
Parquet format has a tag to record if a column is sorted (as far as I know, and assuming this tag applies to the whole dataset).

Could this sorting checked (optionally) and recorded?
From an implementation point of view, this check could be done chunk by chunk at recording, and when one chunk is not sorted, it is not necessary any longer to do the check.

```python
# in a very dummy way of writing
sorted=True
for rg in row_groups:
   sorted = sorted and rg.is_sorted()
   write(rg)
```

Knowing if a column is sorted and checking if it is in ascending or descending order (for instance comparing min/max of a row group) allows some speed up when applying row-wise filtering with `>`, `>=`, `<`, `<=` with respect to a threshold.

Basically, if you do a row-wise filtering (not row-group-wise filtering), you only load the row-groups where the intersection with the threshold occurs (and not all the row groups that intersect with the limit as in current filtering method), and from the index where the intersection occurs, you are able to define `sel` boolean array completely.

I am not so sure the overcost at writing is huge, and the saving can be interesting at reading (avoid to load row-groups).
Bests

**Comment by yohplala** [comment]
PS: i reworte fully the ticket. the story with sorted / not sorted column was an error. Sorry.

**Comment by martindurant** [comment]
This was certainly the intention!

**Comment by martindurant** [comment]
Can you prove that the row-groups filtering is *not* working when row-filtering is active? You might be in a good position to fix this, in that case.

**Comment by yohplala** [comment]
Hi Martin,
I am not sure to understand your comment, but yes, I ought to have some free time available by end of year to propose implementations for some of the tickets I submitted so far. Have a good day, bests!
